deltachat-ffi/src/lib.rs

```diff
@@ -498,11 +467,8 @@ pub unsafe extern "C" fn dc_send_msg(
     chat_id: u32,
     msg: *mut dc_msg_t,
 ) -> u32 {
-    if context.is_null() || chat_id == 0 || msg.is_null() {
```

Comment:
I believe the problem is with this check for `chat_id == 0`.

Python code deliberately passes zero `chat_id` to `dc_send_msg`:
https://github.com/deltachat/deltachat-core-rust/blob/c1a78d9c3ecf5f2de1885f3948a741b18d7a65fe/python/src/deltachat/chatting.py#L214-L215

Reply:
good catch! in fact, chat_id may be 0 if dc_prepare_msg() was called before

Reply:
probably a copy+paste error. careless coding from my side i'd say. sorry.